src/store/actions.ts

```diff
@@ -1105,7 +1105,11 @@ export const startAnalyzeName = async ({ commit, getters }) => {
   // xpro get name call
   if (getters.isXproFlow) {
     commit('mutateXproJurisdiction', getters.getJurisdictionText)
-    commit('mutateHomeJurisNum', getters.getCorpSearch)
+    // set home juris num only if we have a corp num
+    // (don't set if we entered a corp name instead)
+    if (!getters.getHasNoCorpNum) {
+      commit('mutateHomeJurisNum', getters.getCorpSearch)
+    }
```

Comment:
As this does not save the value if we entered a MRAS corp name instead of a MRAS corp num, no invalid corp num can be saved (since it is searched before being accepted).